src/ggl-setup.py

```diff
@@ -0,0 +1,524 @@
+        print(f"Command failed: {e.cmd}. Return code: {e.returncode}")
+        return False
+    except Exception as e:
+        print(f"Error when deleting user and group: {str(e)}")
+        return False
+
+
+def _check_group_exists(group: str) -> bool:
+    try:
+        grp.getgrnam(group)
+        return True
+    except KeyError:
+        return False
+
+
+def _check_user_exists(user: str) -> bool:
+    try:
+        pwd.getpwnam(user)
+        return True
+    except KeyError:
+        return False
+
+
+def _build_with_cmake() -> bool:
+    original_dir = os.getcwd()
+    try:
+        # CMake configure command
+        cmake_cmd = [
+            'cmake', '-B', 'build', '-D', 'CMAKE_INSTALL_PREFIX=/usr/local',
+            '-D', 'CMAKE_BUILD_TYPE=MinSizeRel', '-D', 'CMAKE_BUILD_TYPE=Debug',
+            '-DGGL_LOG_LEVEL=DEBUG'
+        ]
+
+        subprocess.run(cmake_cmd, check=True)
+
+        # Change the directory to build
+        os.chdir("build")
+        build_cmd = ['make', '-C', 'build', '-j$(nproc)']
+        subprocess.run(build_cmd, shell=True, check=True)
+        os.chdir(original_dir)
+
+        # Install command
+        install_cmd = ['sudo', 'make', '-C', 'build', 'install']
+        subprocess.run(install_cmd, check=True)
+
+        print("Successfully completed the build process")
+        return True
+
+    except subprocess.CalledProcessError as e:
+        print(f"Error during build process: {e}")
+        return False
+    except Exception as e:
+        print(f"Unexpected error: {e}")
+        return False
+    finally:
+        os.chdir(original_dir)
+
+
+def _tes_setup(device_cert: str, private_key: str) -> bool:
+
+    # Save the key and certificates
+    try:
+        device_result = _create_file(DEVICE_PATH, device_cert)
+        key_result = _create_file(PRIVATE_PATH, private_key)
+
+        if not device_result or not key_result:
+            return False
+
+        root_ca_url = "https://www.amazontrust.com/repository/AmazonRootCA1.pem"
+        ca_response = requests.get(root_ca_url)
+        ca_result = _create_file(CA_PATH, ca_response.text)
+        if not ca_result:
+            return False
+
+        print("Successfully saved all required certificates and keys")
+        return True
+
+    except Exception as e:
+        print(f"Error when writing certificate or key: {str(e)}")
+        return False
+
+
+def _modify_config(iot_client: client, thing_name: str, file_path: str,
+                   group: str, user: str) -> bool:
+
+    try:
+        iot_data_endpoint = iot_client.describe_endpoint(
+            endpointType='iot:Data-ATS')['endpointAddress']
+        iot_cred_endpoint = iot_client.describe_endpoint(
+            endpointType='iot:CredentialProvider')['endpointAddress']
+
+        with open(file_path, 'r') as file:
+            data = yaml.safe_load(file)
+
+        data['system']['thingName'] = thing_name
+        data['system']['privateKeyPath'] = PRIVATE_PATH
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

 The absolute path starts at the root and moves down through each file. While traversing the path, if the folder name or location is different than the file path mentioned in the code, it causes a break. Since most of the computers have different directory structures, using absolute path in your code is likely to cause issues in your computational environment. We recommend using relative paths, because it only relies on relevant part of the directory structure being the same making it reliable to work with in computational environment

Similar issue at line numbers 360 and 361.